core/src/main/java/io/aiven/klaw/service/OperationalRequestsService.java

```diff
@@ -0,0 +1,341 @@
+      operationalRequestModel.setRequestStatus(RequestStatus.of(topicReq.getRequestStatus()));
+      operationalRequestModel.setOperationalRequestType(topicReq.getOperationalRequestType());
+
+      operationalRequestModel.setTeamname(
+          manageDatabase.getTeamNameFromTeamId(tenantId, topicReq.getRequestingTeamId()));
+
+      // show approving info only before approvals
+      if (RequestStatus.APPROVED != operationalRequestModel.getRequestStatus()) {
+        operationalRequestModel.setApprovingTeamDetails(
+            updateApproverInfo(
+                userList,
+                manageDatabase.getTeamNameFromTeamId(tenantId, userTeamId),
+                approverRoles,
+                operationalRequestModel.getRequestor()));
+      }
+      operationalRequestModelList.add(setRequestorPermissions(operationalRequestModel, userName));
+    }
+    return operationalRequestModelList;
+  }
+
+  private OperationalRequestsResponseModel setRequestorPermissions(
+      OperationalRequestsResponseModel req, String userName) {
+    log.debug(
+        " My request Status {} and userName {} and requestor {}",
+        req.getRequestStatus(),
+        userName,
+        req.getRequestor());
+    if (RequestStatus.CREATED == req.getRequestStatus()
+        && userName != null
+        && userName.equals(req.getRequestor())) {
+      req.setDeletable(true);
+      req.setEditable(true);
+    }
+
+    return req;
+  }
+
+  private String updateApproverInfo(
+      List<UserInfo> userList, String teamName, List<String> approverRoles, String requestor) {
+    StringBuilder approvingInfo = new StringBuilder("Team : " + teamName + ", Users : ");
+
+    for (UserInfo userInfo : userList) {
+      if (approverRoles.contains(userInfo.getRole())
+          && !Objects.equals(requestor, userInfo.getUsername())) {
+        approvingInfo.append(userInfo.getUsername()).append(",");
+      }
+    }
+
+    return String.valueOf(approvingInfo);
+  }
+
+  private List<OperationalRequest> getOperationalRequestsPaged(
+      List<OperationalRequest> origActivityList, String pageNo, String currentPage) {
+
+    List<OperationalRequest> newList = new ArrayList<>();
+    Env envSelected;
+
+    if (origActivityList != null && origActivityList.size() > 0) {
+      int totalRecs = origActivityList.size();
+      int recsPerPage = 10;
+      int totalPages = totalRecs / recsPerPage + (totalRecs % recsPerPage > 0 ? 1 : 0);
+
+      pageNo = commonUtilsService.deriveCurrentPage(pageNo, currentPage, totalPages);
+      int requestPageNo = Integer.parseInt(pageNo);
+      int startVar = (requestPageNo - 1) * recsPerPage;
+      int lastVar = (requestPageNo) * (recsPerPage);
+
+      List<String> numList = new ArrayList<>();
+      commonUtilsService.getAllPagesList(pageNo, currentPage, totalPages, numList);
+
+      for (int i = 0; i < totalRecs; i++) {
+        OperationalRequest operationalRequest = origActivityList.get(i);
+        if (i >= startVar && i < lastVar) {
```

Comment:
why do we need to iterate like `int i = 0; i < totalRecs;` if in fact we need to do something for this condition
`int i = startVar; i < lastVar;` ?

Reply:
Agree, good one.

Reply:
Then we could tweak it a bit
i think this could work `int i = Math.max(0, startVar); i < Math.min(lastVar, totalRecs);`

Reply:
Updated. We have similar logic in almost 10 classes. Would be nice if we can generify it

Reply:
yes, that makes sense, 
probably in a separate PR